packages/expo-splash-screen/ios/EXSplashScreen/EXSplashScreenController.m

```diff
@@ -0,0 +1,107 @@
+// Copyright © 2018 650 Industries. All rights reserved.
+
+#import <EXSplashScreen/EXSplashScreenController.h>
+#import <UMCore/UMDefines.h>
+
+@interface EXSplashScreenController ()
+
+@property (nonatomic, weak) UIViewController *viewController;
+@property (nonatomic, strong) UIViewController *splashScreenViewController;
+@property (nonatomic, strong) UIView *splashScreenView;
+@property (nonatomic, strong) Class rootViewClass;
+@property (nonatomic, weak) UIView *rootView;
+@property (nonatomic, strong) dispatch_queue_t queue;
+
+@property (nonatomic, assign) BOOL autoHideEnabled;
+@property (nonatomic, assign) BOOL splashScreenShown;
+@property (nonatomic, assign) BOOL appContentAppeared;
+
+@end
+
+@implementation EXSplashScreenController
+
+- (instancetype)initWithViewController:(UIViewController *)viewController
+                            resizeMode:(EXSplashScreenImageResizeMode)resizeMode
+              splashScreenViewProvider:(id<EXSplashScreenViewProvider>)splashScreenViewProvider
+{
+  if (self = [super init]) {
+    _viewController = viewController;
+    _autoHideEnabled = YES;
+    _splashScreenShown = NO;
+    _appContentAppeared = NO;
+    _queue = dispatch_queue_create("io.expo.splashscreen", NULL);
```

Comment:
Should this really create a new queue for every controller?

Reply:
We don;'t actually need that queue 👍